corehq/apps/user_importer/importer.py

```diff
@@ -774,24 +780,32 @@ def create_or_update_web_users(upload_domain, user_specs, upload_user, update_pr
 
 
 def modify_existing_user_in_domain(upload_domain, domain, domain_info, location_codes, membership,
-                                   role_qualified_id, upload_user, current_user, max_tries=3):
+                                   role_qualified_id, upload_user, current_user, user_change_logger,
+                                   max_tries=3):
+
     if domain_info.can_assign_locations and location_codes is not None:
         location_ids = find_location_id(location_codes, domain_info.location_cache)
         locations_updated, primary_loc_removed = check_modified_user_loc(location_ids,
                                                                          membership.location_id,
                                                                          membership.assigned_location_ids)
         if primary_loc_removed:
             current_user.unset_location(domain, commit=False)
+            user_change_logger.add_changes({'location_id': current_user.location_id})
+            if current_user.location_id:
+                user_change_logger.add_info(_(f"Primary location: {current_user.get_sql_location(domain).name}"))
```

Comment:
this was later moved to be after the `if locations_updated:` block because the primary location can get re-set in that section.
https://github.com/dimagi/commcare-hq/pull/29935/commits/ebad364ef0537a2a96220efe60a770072d372b6c